scripts/clean_validate.py

```diff
@@ -0,0 +1,44 @@
+# clean_validate.py
+# author: Weilin Han
+# date: 2024-10-20
+
+import click
+import os
+import sys
+import pandas as pd
+import pandera as pa
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from src.clean_data import extract_column_name, read_raw_data, clean_data, write_data
+from src.validate_data import build_schema_from_csv, validate_data
+
+@click.command()
+@click.option('--raw-data-file', type=str, help="Path to raw data file")
+@click.option('--name-file', type=str, help="Path to names file")
+@click.option('--data-config-file', type=str, help="Path to data configuration file")
+@click.option('--write-to', type=str, help="Path to directory where cleaned data will be written to")
+@click.option('--written-file-name', type=str, help="The name of the file will be written")
+
+def main(raw_data_file, name_file, data_config_file, write_to, written_file_name):
+    """Clean raw data and validate it."""
+    # Extract column names from .names file
+    colnames = extract_column_name(name_file)
+
+    # Read raw data
+    imported_data = read_raw_data(raw_data_file, colnames)
+
+    # Removing id column and relabel diagnosis column
+    cleaned_data = clean_data(imported_data)
+
+    
+    # Create schema
+    config_df = pd.read_csv(data_config_file)
+    
+    schema=build_schema_from_csv(data_config=config_df, expected_columns=colnames[1:]) #removing id colnames list
```

Comment:
Let's use the column name to drop the ID column name from the list. Numerical indexing makes code brittle and is less readable.
```suggestion
    clean_colnames = colnames.remove("id")
    schema = build_schema_from_csv(data_config=config_df, expected_columns=clean_colnames) 
```

Reply:
Also, wondering if we should rename `build_schema_from_csv` to `build_schema_from_DataFrame` since that is what the function does (like it would work if we created a data frame from `pd.DataFrame` with code and then used the function.